public/scripts/presetScript.js

```diff
@@ -0,0 +1,99 @@
+const presetBox = document.querySelector("#preset-box");
+
+async function savePreset() {
+    // TODO: Add game theme
+    let presetNameInput = document.querySelector("#preset_name");
+    let durationInput = document.querySelector("#duration");
+    let intervalInput = document.querySelector("#interval");
+    let lootLats = [];
+    let lootLngs = [];
+    let borderLats = [];
+    let borderLngs = [];
+
+    let mapDataValid = (lootLatLngs.length > 0 && markerLatLngs.length > 0 && policeStationLatLng != null);
+    let durationInputValid = (durationInput.value && durationInput.checkValidity());
+    let intervalInputValid = (intervalInput.value && intervalInput.checkValidity());
+    let presetNameValid = (presetNameInput.value && presetNameInput.value.trim() !== "");
+
+    if (!mapDataValid) {
+        showValidationError('Plaats a.u.b. alle pins op de kaart.');
+        return;
+    }
+    if (!durationInputValid) {
+        showValidationError('Vul a.u.b. een valide speelduur in.');
+        return;
+    }
+    if (!intervalInputValid) {
+        showValidationError('Vul a.u.b. een valide interval in tussen locatieupdates.');
+        return;
+    }
+    if (!presetNameValid) {
+        showValidationError('Vul a.u.b. een naam in voor het template.');
+        return;
+    }
+
+    lootLatLngs.forEach(lootItem => {
+        lootLats.push(lootItem.lat);
+        lootLngs.push(lootItem.lng);
+    });
+    markerLatLngs.forEach(border => {
+        borderLats.push(border.lat);
+        borderLngs.push(border.lng);
+    });
+
+    $.ajaxSetup({
+        headers: {
+            'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+        }
+    });
+    await $.ajax({
+        url: '/presets',
+        type: 'POST',
+        data: {
+            name: presetNameInput.value,
+            duration: durationInput.value,
+            interval: intervalInput.value,
+            police_station_lat: policeStationLatLng.lat,
+            police_station_lng: policeStationLatLng.lng,
+            loot_lats: lootLats,
+            loot_lngs: lootLngs,
+            loot_names: lootNames,
+            border_lats: borderLats,
+            border_lngs: borderLngs
+        },
+        error: function (err) {
+            console.log(err);
+        }
+    });
+}
+
+function loadPreset() {
+
+}
+
```

Comment:
```suggestion
```

Reply:
Zelfde verhaal als eerder.

Reply:
volledige implementatie staat in mijn PR